tools/pd-ctl/tests/global_test.go

```diff
@@ -16,34 +16,51 @@ package tests
 
 import (
 	"context"
-	"fmt"
+	"encoding/json"
 	"net/http"
 	"testing"
 
+	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/log"
 	"github.com/stretchr/testify/require"
 	"github.com/tikv/pd/pkg/utils/apiutil"
 	"github.com/tikv/pd/pkg/utils/assertutil"
 	"github.com/tikv/pd/pkg/utils/testutil"
 	"github.com/tikv/pd/server"
+	"github.com/tikv/pd/server/api"
 	cmd "github.com/tikv/pd/tools/pd-ctl/pdctl"
+	"github.com/unrolled/render"
 	"go.uber.org/zap"
 )
 
-const pdControlCallerID = "pd-ctl"
+const pdControlCallerID = "pd-http-client"
 
 func TestSendAndGetComponent(t *testing.T) {
 	re := require.New(t)
 	handler := func(context.Context, *server.Server) (http.Handler, apiutil.APIServiceGroup, error) {
 		mux := http.NewServeMux()
+		mux.HandleFunc("/pd/api/v1/cluster", func(w http.ResponseWriter, _ *http.Request) {
+			cluster := &metapb.Cluster{
+				Id: 0,
+			}
+			var rd render.Render
+			rd.JSON(w, http.StatusOK, cluster)
+		})
 		mux.HandleFunc("/pd/api/v1/health", func(w http.ResponseWriter, r *http.Request) {
 			callerID := apiutil.GetCallerIDOnHTTP(r)
 			for k := range r.Header {
 				log.Info("header", zap.String("key", k))
 			}
 			log.Info("caller id", zap.String("caller-id", callerID))
 			re.Equal(pdControlCallerID, callerID)
-			fmt.Fprint(w, callerID)
+			healths := []api.Health{
+				{
+					Name: "test",
+				},
+			}
+			healthsBytes, err := json.Marshal(healths)
+			re.NoError(err)
+			w.Write(healthsBytes)
```

Comment:
It seems like this test wants to focus on `calledID`, so I think it's no need to change it.

FYI https://github.com/tikv/pd/pull/4491/files#diff-05102750a57e22445f5760fb1545227c9130a3a093f3b879e1f6e48ce27ea890

Reply:
L55 has tested it. If we keep returning `calledID`, json.UnMarshal will be failed.

Reply:
https://github.com/tikv/pd/blob/4cd42b3cb8e7ae32c4d15d6fab7d7ab0552facf2/client/http/client.go#L310

The HTTP client created in the `pd-ctl` is assigned by `defaultCallerID`. Do we need to change it to `source`. `source` equals to "pd-ctl" string.